pkg/common/utils_test.go

```diff
@@ -1220,6 +1228,29 @@ func TestCodeForError(t *testing.T) {
 			inputErr: &googleapi.Error{Code: http.StatusBadRequest, Message: "User error with bad request"},
 			expCode:  codes.InvalidArgument,
 		},
+		{
+			name: "googleapi.Error that wraps apierror.APIError, which wraps a googleapi.Error",
+			inputErr: getAPIWrappedError(&googleapi.Error{
+				Code:    403,
+				Message: "Some permission error",
+			}, true /* wrap */),
+			expCode: codes.PermissionDenied,
+		},
+		{
+			name: "googleapi.Error that wraps apierror.APIError of http kind",
+			inputErr: getAPIWrappedError(&googleapi.Error{
+				Code:    404,
+				Message: "data requested not found error",
+			}, false /* wrap */),
+			expCode: codes.NotFound,
+		},
+		{
+			name: "googleapi.Error that wraps apierror.APIError of status kind",
```

Comment:
Can you also add a testcase for:
1. apierror.APIError wraps googleAPI error. 
2. apierror.APIError wraps status error. 

I know you have test cases for googleapi that wraps 1, and 2, but I think adding these without the outer wrapping of googleapi error is a good idea.